### PR TITLE
feat: support influxdb

### PR DESCRIPTION
**Pull request by lucsorel**
Thanks to the team for this great testing utility :pray: 

This PR brings the possibility to spawn a Docker instance of InfluxDB (timeseries-oriented database) for integration testing. This PR supports both 1.x and 2.x versions of InfluxDB, which rely on different query languages and require the use of different Python client libraries. Therefore:

- there is a root `InfluxDbContainer` class in `testcontainers/influxdb.py` for the common mechanisms (health check, connection url, etc.)
- there are 2 separate classes, one for each InfluxDB major version: `InfluxDb1Container` class in `testcontainers/influxdb1/__init__.py`, `InfluxDb2Container` in `testcontainers/influxdb2/__init__.py` that are meant to be used by the testcontainers users. Each one has its own `.get_client()` method involving the dedicated Python client library

Unit and integration tests (meant to be examples for the testcontainers end-users) are provided in the PR.

I declared the 2 Python client libraries as dependencies in `setup.py` so that the continuous integration of testcontainers can be launched. I wonder whether it implies that both clients will be installed by the testcontainers end-users although they are likely to need only one: either the 1.x client or the 2.x client. Installing the 2 but using only 1 works, but it slows down the installation and test processes. I am open to advice from the maintainers for that :pray:

### CODE REVIEW

**Comment by lucsorel** [comment]
in the meantime, I created the https://github.com/Purecontrol/testcontainers-python-influxdb to publish the https://pypi.org/project/testcontainers-python-influxdb/ package.

**Comment by alexanderankin** [comment]
please review:

https://github.com/testcontainers/testcontainers-python/compare/main...alexanderankin:testcontainers-python:feat/support-influxdb

fixed a test here:
* https://github.com/testcontainers/testcontainers-python/commit/2e219285e6db2afd6c4aaae9c44bdccb7043f06b
* https://github.com/testcontainers/testcontainers-python/compare/main...alexanderankin:testcontainers-python:feat/support-influxdb_adjust

**Comment by lucsorel** [comment]
hello @alexanderankin

Thank you for the feedback on this topic :pray: 

I am sorry, I am not sure to understand what you want me to do :disappointed: most of the links you sent me are links to create pull requests on your github account. Don't you want to create them first and then ask me to review them?

Or would you like me to "cherry-pick" the changes on the PR I opened?

**Comment by alexanderankin** [comment]
Sorry I should have clarified - can you confirm that I did not make any
unintentional changes while rebasing? I would force push but I think it's
impolite without first asking

**Comment by alexanderankin** [comment]
If it's okay with you I will replace your 9 original commits with the seven commits which result from rebasing onto the new structure (two of them edited files which no longer exist, such as setup.py)

**Comment by lucsorel** [comment]
thank you for clarifying and asking.

> If it's okay with you I will replace your 9 original commits with the seven commits which result from rebasing onto the new structure (two of them edited files which no longer exist, such as setup.py)

Yes, that is fine :+1:

**Comment by lucsorel** [comment]
there is just one change that should be done: those lines (https://github.com/testcontainers/testcontainers-python/compare/main...alexanderankin:testcontainers-python:feat/support-influxdb_adjust#diff-d05626efd4466a0f9ddf3973c13071ea2e534dc69a295e68cc9fc3e5a5541e0fR115-R145) should be removed because they force the main container class to import the dependencies for InfluxDB version 1.x and 2.x, whereas users may use either one or the other, but it is not a good idea to force them to install the dependencies of both versions.

Instead, both [InfluxDb1Container](https://github.com/testcontainers/testcontainers-python/compare/main...alexanderankin:testcontainers-python:feat/support-influxdb_adjust#diff-bd64360629a8b2ac551283a6d10500059bfb36e17340e484bc1d8fd0322c81ccR47-R63) and [InfluxDb2Container](https://github.com/testcontainers/testcontainers-python/compare/main...alexanderankin:testcontainers-python:feat/support-influxdb_adjust#diff-e5cc1b8aa3516d4eff82f2f95976f6aeeb926b75a12a7d8aea92a2a043a287aaR82-R102) classes are independent and have their own `get_client()` method. 

Once you have force-pushed, your changes, I can add a commit to remove those unnecessary lines and imports, if you like?

**Comment by alexanderankin** [comment]
Yes feel free to yourself as well as I will be at work. Since the commits
are pushed to GitHub public repo you have access to everything

**Comment by alexanderankin** [comment]
actually im not sure where the `influxdb/__init__.py` came from - maybe this is an artifact of a bad rebase - seems more cleanup is necessary as well

**Comment by alexanderankin** [comment]
figured it out, fixing

**Comment by alexanderankin** [comment]
fixed

**Comment by alexanderankin** [comment]
feel free to layer your changes on this i rebased to address the part about the client_v1 and v2 so its in your repo now where you can work with it normally. feel free to run the "pre-commit run -a" to test linting locally as well

**Comment by lucsorel** [comment]
thank you for fixing the `client_v1` and `_v2`.

I don't have anything to add, but I can easily fix the linting issues if you like.

**Comment by lucsorel** [comment]
I see that testcontainers-python now uses pre-commit to run black for formatting and ruff for linting. I would suggest to use ruff-format as a drop-in replacement for black, but maybe in another issue?

**Comment by lucsorel** [comment]
I pushed the commit related to the pre-commit hooks and the lint hooks now passes :+1: 

I am just wondering how to launch the influxdb unit tests locally. I believe the steps are:
1. install the optional dependencies for influxdb only
2. run the tests for influxdb only
I am not sure how to do that, now that the codebase has been moved to the `modules` folder

**Comment by lucsorel** [comment]
`poetry install -E influxdb` seems to do the job for the dependencies

**Comment by alexanderankin** [comment]
Poetry install -E arangodb

Poetry shell
Pytest modules/arangodb

Poetry run pytest/modules/arangodb

**Comment by lucsorel** [comment]
`poetry run pytest modules/influxdb` worked indeed :pray: 

I pushed a fix on the unit test. I the pipeline succeeds, I believe we're good to approve and merge the PR :smiley:

**Comment by lucsorel** [comment]
I would suggest to add the documentation about running the automated tests of a given module either in the root `README.md` or in `modules/README.md`. What do you think?

File: modules/influxdb/README.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+.. autoclass:: testcontainers.influxdb.InfluxDbContainer
```
Comment (alexanderankin) [inline modules/influxdb/README.rst:1]: Needs title like other modules

**Comment by alexanderankin** [comment]
Ok lgtm then

**Comment by lucsorel** [comment]
> Ok lgtm then

I think a maintainer needs to trigger the CI workflow to validate all the changes (see https://github.com/testcontainers/testcontainers-python/actions) and the PR. Moreover, I cannot approve the PR myself.

**Comment by lucsorel** [comment]
all the checks have passed successfully :smiley:

**Comment by lucsorel** [comment]
I resolved a conflict on the `poetry.lock` lockfile. The CI workflows need to be relaunched

**Comment by lucsorel** [comment]
all checks have passed :tada:  thank you :pray: 

Is there an action that I should do for the merge?

**Pull request merged**